src/lib/comms/include/sol-bluetooth.h

```diff
@@ -0,0 +1,90 @@
+/*
+ * This file is part of the Soletta Project
+ *
+ * Copyright (C) 2016 Intel Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <sol-network.h>
+
+enum sol_bt_uuid_type {
+	SOL_BT_UUID_TYPE_16 = 2,
+	SOL_BT_UUID_TYPE_32 = 4,
+	SOL_BT_UUID_TYPE_128 = 16,
+};
+
+struct sol_bt_uuid {
+	enum sol_bt_uuid_type type;
+	union {
+		uint16_t val16;
+		uint32_t val32;
+		uint8_t val128[16];
+		uint8_t val[0];
+	};
+};
+
+int sol_bt_uuid_from_str(struct sol_bt_uuid *uuid, const char *str);
+int sol_bt_uuid_to_str(const struct sol_bt_uuid *uuid, struct sol_buffer *buffer);
+
+struct sol_bt_conn;
+
+struct sol_bt_conn *sol_bt_conn_ref(struct sol_bt_conn *conn);
+void sol_bt_conn_unref(struct sol_bt_conn *conn);
+
+struct sol_bt_device_info {
+	struct sol_network_link_addr addr;
+	char *name;
+	struct sol_vector uuids;
+	int16_t rssi;
+};
+
+int sol_bt_set_connection_callbacks(void *(connected)(void *user_data,
+						      struct sol_bt_conn *conn),
+				    void *(disconnected)(void *user_data,
+							 struct sol_bt_conn *conn),
+				    const void *user_data);
```

Comment:
I don't know. But I am having a slight preference for the suggestion I made above. I will try to produce a sample that exercises this and see how it would look.